### PR TITLE
Add migration check for nodes when typha is deployed

### PR DESCRIPTION
**Pull request by tmjd**
Make sure enough nodes exist for the current typha and new typha

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/migration/convert/typha.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/controller/utils"
+)
+
+func checkTypha(c *components, _ *operatorv1.Installation) error {
+	if c.typha != nil {
```
Comment (ozdanborne) [inline pkg/controller/migration/convert/typha.go:13]: nit, you can de-indent this if you return
```suggestion
	if c.typha == nil {
	  return nil
	}
```

File: pkg/controller/migration/convert/typha.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		}
+
+		curReplicas := int(*c.typha.Spec.Replicas)
+
+		if curReplicas == 0 {
+			return nil
+		}
+
+		nodes := corev1.NodeList{}
+		err := c.client.List(ctx, &nodes)
```
Comment (ozdanborne) [inline pkg/controller/migration/convert/typha.go:25]: wild idea: should we avoid this client request by checking the status of the existing calico-node daemonset to see how many instances it's running?

**Review by ozdanborne — APPROVED** [review]
couple optional nits but lgtm

File: pkg/controller/migration/convert/typha.go (hunk 2 above)
Comment (tmjd) [reply pkg/controller/migration/convert/typha.go:25]: That is an interesting idea. Though I think there would be issues if operator restarted near when there were only a few of the 'old' calico-node pods left.

**Pull request merged**